src/main/java/com/google/devtools/build/lib/profiler/TraceEvent.java

```diff
@@ -57,6 +63,8 @@ public static TraceEvent create(
   // Only applicable to action-related TraceEvents.
   @Nullable
   public abstract String primaryOutputPath();
+  @Nullable
+  public abstract Map<String, Object> args();
```

Comment:
Instead of storing and exposing the full map here, I would only expose what we know of, in this case the action owner.
So parsing the full map below is fine and then extracting the data that we need.

Reply:
My understanding of that object was that it closely mimics the structure of the trace file record, but I can change to proposed no problem.